### ISSUE TITLE
Bower 1.5.3 doesn't upgrade/downgrade if version is changed in bower.json

### ISSUE THREAD

**Issue by darkmavis1980** [bower/bower#1934]
I just noticed that if a new version of a package is release, or even if I downgrade a package to a previous version, bower doesn't remove the previously installed version and keep using the package previously installed.
The only way to downgrade/upgrade is to delete the package from the bower_components folder and run then run the `bower install` command.

**Comment by sheerun** [comment]
I can't reproduce on 1.6.2. Could you post exact steps to reproduce?

**Comment by darkmavis1980** [comment]
Imagine that I've previously installed with bower the package `xyz@1.0.0`, if a newer version was available, like `1.0.1`, even if I clearly state on the `bower.json` to get the newest version of the package, and even if clear the bower cache, when I perform a `bower update` it will not perform an update and it will keep the version `1.0.0`.
The only way to fix it is to manually delete the package in the **bower_components** folder and then run bower update/install

**Comment by sheerun** [comment]
Here's how I'm unable to reproduce:

```
$ bower install --save angular#1.4.6
$ cat bower_components/angular/package.json | grep version
  "version": "1.4.6",
$ // edit bower.json and change angular to 1.4.7
$ bower update
$ cat bower_components/angular/package.json | grep version
  "version": "1.4.7",
```

The same happens with `bower install` instead of `bower update`. I'm testing on bower 1.6.2.

Can you give me set of commands and bower version to reproduce this error?

**Comment by darkmavis1980** [comment]
I'm doing the same thing, but did you test it with bower **1.5.3**?

**Comment by sheerun** [comment]
Yes, the behaviour is still the same in bower 1.5.3. Please try `npm cache clean` and `bower cache clean`, and then run these steps again? From empty bower.json? What is your platform?

**Comment by darkmavis1980** [comment]
Hi @sheerun, sorry for the late reply, I could take a look at it.
Since the bug I've updated my local env to the latest version of bower and now it's fine, I tried to replicate the issue on a VM I've created but the `update` works fine there, so it's fine.
It was odd because what I saw then on my local machine, and on a couple of other machines, is that even if I said on the `bower.json` file to get the latest version of Angular 1.4 with this:
```
"dependencies": {
    "angular": "~1.4.x"
  }
```
which supposed to auto-update if the local version was out of date, it didn't update anything.
Anyway, the issue seems to be resolved, so it's better to mark this bug as closed.
